packages/belos/src/BelosPseudoBlockGmresSolMgr.hpp

```diff
@@ -758,32 +779,52 @@ setParameters (const Teuchos::RCP<Teuchos::ParameterList>& params)
 
     if (tempOrthoType != orthoType_) {
       orthoType_ = tempOrthoType;
+
       params_->set("Orthogonalization", orthoType_);
+
+      if (haveOrthoSublist) {
+        params_->set("OrthoManager", *orthoParams);
+      }
+
       // Create orthogonalization manager
       if (orthoType_ == "DGKS") {
         typedef DGKSOrthoManager<ScalarType, MV, OP> ortho_type;
         if (orthoKappa_ <= 0) {
-          ortho_ = rcp (new ortho_type (label_));
```

Comment:
Ok, all of this additional logic for processing the orthogonalization type and parameters I think should be relegated to the orthogonalization manager.  This logic block will be repetitive among all iterative solvers that require orthogonalization.  Furthermore, I do not know of anyone who actually changes the "Orthogonalization Constant" from DGKS.  So, I would be ok just removing that logic from all solver managers and moving the parameter into the sublist.